odoo/addons/base/tests/test_expression.py

```diff
@@ -106,6 +106,17 @@ def test(op, value, expected):
         test('not ilike', 'B', ['0', 'a'])
         test('not like', 'AB', ['0', 'a', 'b', 'a b'])
 
+    def test_09_hierarchy_filtered_domain(self):
+        Partner = self.env['res.partner']
+
+        Partner = self.env['res.partner']
+        p = Partner.create({'name': 'dummy'})
+
+        # hierarchy without parent
+        self.assertFalse(p.parent_id)
+        self._search(Partner, [('parent_id', 'parent_of', p.id)], [('id', '=', p.id)])
+        self._search(Partner, [('parent_id', 'child_of', p.id)], [('id', '=', p.id)])
```

Comment:
Both should be equal to `p`, right?  Maybe you should assert that as well.

Reply:
Good point.